utilities/blob_db/blob_compaction_filter.cc

```diff
@@ -360,11 +360,27 @@ CompactionFilter::BlobDecision BlobIndexCompactionFilterGC::PrepareBlobOutput(
 
   PinnableSlice blob;
   CompressionType compression_type = kNoCompression;
+  std::string compression_output;
   if (!ReadBlobFromOldFile(key, blob_index, &blob, false, &compression_type)) {
     gc_stats_.SetError();
     return BlobDecision::kIOError;
   }
 
+  // If the compression_type is changed, re-compress it with the new compression
+  // type.
+  if (compression_type != blob_db_impl->bdb_options_.compression) {
+    if (compression_type != kNoCompression) {
+      const Status status =
+          blob_db_impl->DecompressSlice(blob, compression_type, &blob);
+      if (!status.ok()) {
+        return BlobDecision::kCorruption;
```

Comment:
We should call `gc_stats_.SetError()` here, like in the other failure cases (note: this is not needed if we make the change below).

Reply:
Just checking: are you planning to rely on the `ReadBlobFromOldFile` call above to do the decompression?